signature-aggregator/aggregator/aggregator.go

```diff
@@ -382,14 +422,37 @@ func (s *SignatureAggregator) handleResponse(
 		return nil, true, nil
 	}
 
+	if signedMsg, err := s.aggregateIfSufficientWeight(
+		unsignedMessage,
+		signatureMap,
+		accumulatedSignatureWeight,
+		connectedValidators,
+		quorumPercentage,
+	); err != nil {
+		return nil, true, err
+	} else if signedMsg != nil {
+		return signedMsg, true, nil
+	}
+
+	// Not enough signatures, continue processing messages
+	return nil, true, nil
```

Comment:
I generally dislike the inlined `if` when the function returns more than just `err`.

Reply:
Happy to defer to team opinion, but... in general I find the inlined `if` syntax weird and generally avoid it, but here I wrote it this way in order to limit the scope of `signedMsg` so that it's more clear to the reader that this instance of `signedMsg` _only_ refers to the one that's returned here, and has nothing to do with the one that's returned from `handleResponse` later in this function.